### PR TITLE
Figure.meca: Expand docstring for the scale parameter

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to expand the docstring for the `scale` parameter of `pygmt.Figure.meca`.

**Upstream GMT documentation**: https://docs.generic-mapping-tools.org/dev/supplements/seis/meca.html#s

**Preview**: https://pygmt-dev--2552.org.readthedocs.build/en/2552/api/generated/pygmt.Figure.meca.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/meca.py [hunk 1]
```diff
@@ -159,11 +159,23 @@ def meca(
           ``convention`` parameter is required so we know how to interpret the
           columns. If ``spec`` is a dictionary or a pd.DataFrame,
           ``convention`` is not needed and is ignored if specified.
-
     scale : str
-        Adjust the scaling of the radius of the beachball, which is
-        proportional to the magnitude. *scale* defines the size for
-        magnitude = 5 (i.e. scalar seismic moment M0 = 4.0E23 dynes-cm).
+        *scale*\ [**+a**\ *angle*][**+f**\ *font*][**+j**\ *justify*]\
+        [**+l**][**+m**][**+o**\ *dx*\ [/\ *dy*]][**+s**\ *reference*]
```
Comment (yvonnefroehlich) [inline pygmt/src/meca.py:164]:
```suggestion
        *scale*\ [**+a**\ *angle*][**+f**\ *font*][**+j**\ *justify*]\
        [**+l**][**+m**][**+o**\ *dx*\ [/\ *dy*]][**+s**\ *reference*].
```

**Comment by seisman** [comment]
This PR also fixes https://github.com/GenericMappingTools/pygmt/issues/1466.

**Review by seisman — APPROVED** [review]
Looks good to me. We can have a more Pythonic `Figure.meca` function later if we want.

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers to review this PR.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**